clang/lib/AST/Type.cpp

```diff
@@ -5030,6 +5030,29 @@ bool Type::hasSizedVLAType() const {
   return false;
 }
 
+bool Type::isHLSLIntangibleType() const {
+  const Type *Ty = getUnqualifiedDesugaredType();
+
+  // check if it's a builtin type first (simple check, no need to cache it)
```

Comment:
> no need to cache it

This suggests to me that we cache some of the other results?  Is that correct?

Oh, I see that this is just moved code, may be worth cleaning the comment up at some point if it is just muddying the water.

Reply:
I've updated the comment.